src/csync/csync_reconcile.cpp

```diff
@@ -310,10 +344,19 @@ static int _csync_merge_algorithm_visitor(csync_file_stat_t *cur, CSYNC * ctx) {
                 break;
             case CSYNC_INSTRUCTION_IGNORE:
                 cur->instruction = CSYNC_INSTRUCTION_IGNORE;
-            break;
+                break;
             default:
                 break;
             }
+            // Ensure we're not leaving discovery-only instructions
+            // in place. This can happen, for instance, when other's
+            // instruction is EVAL_RENAME because the parent dir was renamed.
+            // NEW is safer than EVAL because it will end up with
```

Comment:
I guess you meanst s/EVAL/SYNC/g  (also next line)

Reply:
No. I mean that changing discovery's EVAL to NEW is generally safe in reconcile, since all `switch` clauses typically check for EVAL/NEW. Switching to SYNC may have side effects that I haven't investigated. The effect for propagation is the same. (Why do we even distinguish NEW/SYNC? For reporting?)